src/generate_tangent.jl

```diff
@@ -0,0 +1,13 @@
+rand_tangent(rng::AbstractRNG, x::Integer) = Zero()
```

Comment:
This is a hard call.

Reply:
Haha yeah. Not sure that I've got this right tbh. Depends whether we want to treat the integers as being embedded in the real line, or as being their own thing 🤷

Reply:
I'm in favor of treating integers as integers. Unless I'm missing something, the only reason to treat integers as embedded in the reals is so that if someone doesn't bother to represent an input to a rule or to a function like `Zygote.gradient` with a float, it still gives them a reasonable answer. Otherwise, integers will be introduced as hard-coded values whose adjoints are unused or quantities with zero derivatives (e.g. produced by `round`).

And then there are cases like the adjoint of `p::Integer` in `y=x^p`. If you treat `p` as embedded in the reals, then the pullback must produce a complex adjoint on `p`, since `x` could be negative, even though for integer `p`, the function takes reals and produces a real. But if you treat `p` as an integer, then it has an adjoint of `Zero()`. To me, the latter seems more like the right thing than the former.

Reply:
I've doubled down on this now, going with `DoesNotExist`.